avalon/nuke/lib.py

```diff
@@ -0,0 +1,240 @@
+import os
+import contextlib
+import nuke
+import re
+import logging
+import toml
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def maintained_selection():
+    nodes = nuke.allNodes()
+    previous_selection = [n.name()
```

Comment:
Can't you get the selected nodes in nuke with: `nuke.selectedNodes()`? I see it being used [here](https://stackoverflow.com/a/35048809/1838864).

Reply:
Yes!